### PR TITLE
feat(api): add API_GetUserGameAwards endpoint

### PR DESCRIPTION
**Pull request by wescopeland**
This PR adds a new endpoint: _API_GetUserGameAwards.php_.

**Why?**

While it is currently possible to fetch a user's game completion stats with `API_GetUserCompletedGames`, this does not actually neatly map to what game awards a user has. For example, if a game has added new achievements in the months/years since the game was mastered by the user, it is not possible using the existing endpoint to know if a given user actually earned a mastery from the game.

This leads to awkward situations where a Discord bot might announce "User X just unlocked their 25th mastery!", when actually they got their 25th mastery years ago.

**How?**

```
GET https://retroachievements.org/API/API_GetUserGameAwards.php?z=...&y=...&u=SomeUser
```

```ts
[
  {
    AwardedAt: "2019-02-07T22:15:28+00:00",
    GameID: "2145",
    GameName: "R-Type",
    ConsoleID: "8",
    ConsoleName: "PC Engine",
    IsMastery: true,
    DisplayOrder: "0",
    IsMissingNewAchievementUnlocks: false
  },
  // ...
]
```

---

In the interest of not creating technical debt, this PR is in Draft status until I have added tests. I will add tests once https://github.com/RetroAchievements/RAWeb/pull/1357 is merged.

### CODE REVIEW

File: public/API/API_GetUserGameAwards.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+use LegacyApp\Community\Enums\AwardType;
+
+$user = request()->query('u');
+
+$allUserSiteAwards = getUsersSiteAwards($user);
+
+$filteredAwards = [];
+foreach ($allUserSiteAwards as $userSiteAward) {
+  $currentAwardType = (int) $userSiteAward['AwardType'];
```
Comment (luchaos) [inline public/API/API_GetUserGameAwards.php:27]: ℹ️ `.editorconfig` assumes 4 spaces for php files

Comment (wescopeland) [reply public/API/API_GetUserGameAwards.php]: Updated

**Comment by wescopeland** [comment]
I'll refer to https://github.com/RetroAchievements/RAWeb/pull/1396 to see if I can figure out how to write tests for these. Seems like there's a number of great examples circulating now.

**Comment by wescopeland** [comment]
Add validation per https://github.com/RetroAchievements/RAWeb/pull/1402

**Comment by wescopeland** [comment]
Closing for the time being, this uses a bit of code that is currently undergoing refactor. I will rework this when https://github.com/RetroAchievements/RAWeb/pull/1393 is merged.

**Pull request closed without merging**